lib/src/request.dart

```diff
@@ -130,6 +133,28 @@ class Request extends Message {
       {Map<String, String> headers, Map<String, Object> context})
       : this('DELETE', url, headers: headers, context: context);
 
+  factory Request.multipart(url,
+      {Map<String, String> headers,
+        Map<String, Object> context,
+        Map<String, String> fields,
+        Iterable<MultipartFile> files}) {
+    fields ??= <String, String>{};
```

Comment:
You shouldn't need explicit type parameters here; inference should take care of that.

Also, you might as well make these const to avoid unnecessary allocations.